src/vulkan/buffer_descriptor.h

```diff
@@ -31,14 +32,9 @@ namespace vulkan {
 class CommandBuffer;
 class Device;
 
-enum class DescriptorType : uint8_t {
-  kStorageBuffer = 0,
-  kUniformBuffer,
-};
-
 /// Stores descriptor set and binding information for storage and uniform
 /// buffers.
-class BufferDescriptor {
+class BufferDescriptor : public Descriptor {
```

Comment:
How difficult would it be to put the changes for splitting out the Descriptor class into a separate CL which is mostly just moving code around? (Similar for, I think, the image resource code that was split out? We should be able to split and land it pretty quickly.

Reply:
Most likely it sounds easier than it is and would require some kind of intermediate code that would then be discarded for the final result. I'd rather get the storage image support into good shape quickly and then add other descriptor types one by one as separate PRs.